conans/client/cmd/download.py

```diff
@@ -14,9 +14,13 @@ def download(app, ref, package_ids, remote, recipe, recorder, remotes):
     hook_manager.execute("pre_download", reference=ref, remote=remote)
 
     try:
-        ref = remote_manager.get_recipe(ref, remote)
+        if not ref.revision:
+            ref = remote_manager.get_latest_recipe_revision(ref, remote)
     except NotFoundException:
         raise RecipeNotFoundException(ref)
+    else:
+        ref_in_cache = cache.get_recipe_revisions(ref)
```

Comment:
This ``cache.get_recipe_revision(ref)`` when ref contains revision, and that is supposed to return just 1 revision, the passed one, is a bit confusing to read. Maybe is worth to add an explicit ``bool recipe_revision_exist(ref)`?